### DISCUSSION TITLE
Min time between updates (API limits) and wind gusts

### DISCUSSION THREAD

**Discussion by potemy** [cytech/Home-Assistant-wundergroundpws#138] [Q&A] [answered]
I tried to minimize the time between updates (wunderground_data.py) and I've found that 1min is too small value (api limit is over) but 2 min is fine.
So I have question about wind gusts (max values). There is difference in max gusts between station values and HA values. It is probably because this long time between updates. HA download not each pocket but for example every third or fifth data packet from station. Is there any possibility to take from station and send to HA max wind gust not from last packet of data but for example from last 2 minutes if there is time between updates 2 minutes set in "wunderground_data.py"
Max gusts are very urgent for me and this differences are very big.

**Comment by cytech** [comment, marked as answer]
nope
the api is only returning the CURRENT condition (wind_gust) at the time it is accessed, and as you noted you will blowup the api limit if you are trying every minute.
The wunderground personal weather station API does not provide access to historical information.
If you really need to work with realtime (rapid) information, hass/wundergroundpws is not the best way to do it.
You should look at something like WEEWX  https://weewx.com

**Comment by nepozs** [comment]
@potemy 
There is also another alternative (and completely different way of receiving data, but it is possible to receive every packet sent by transmitter) - many models of weather station transmitters of different brands can be used with an SDR receiver (USB Realtek dvb-tuner or dedicated SDR dongle)
https://github.com/pbkhrv/rtl_433-hass-addons

Of course it is possible only if weather station and HA are at the same location.

**Reply by jmtatsch** [reply]
Or you could try the official ecowitt integration. Should work for all fineoffset clones so most weather stations on the market.

**Comment by potemy** [comment]
Yes, it is a good idea to use 433 integration.
Weather station is in the same location and it use the same wifi
I have EUROCHRON EFWS2900 so I dont know if it will work with this integration?
Which one integration you suggest?

I need it only for wind gusts because all other data are ok, and 2min time between updates is enough for all others data...

**Reply by jmtatsch** [reply]
That looks exactly like mine (a fine offset clone). You can use this https://www.home-assistant.io/integrations/ecowitt/ and set a push frequency of lower than 60 s. You won't be able to go faster than the sample rate of the sensors though and I would be surprised if they send more often than once a minute for energy preservation reasons.
